timesketch/frontend/src/components/Common/TsIOCMenu.vue

```diff
@@ -0,0 +1,184 @@
+<!--
+Copyright 2019 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<template>
+  <div style="display:inline;">
+    <span
+      class="ioc-match"
+      ref="contextMenuParent"
+      @click="event => this.$refs.contextMenu.open(event, getIOC($attrs.text), $refs.contextMenuParent)"
+    >
+      <slot></slot>
+    </span>
+    <TsContextMenu ref="contextMenu">
+      <template v-slot="params">
+        <section v-if="params.data" class="box ioc-context-menu">
+          <div class="ioc-display">
+            <span
+              class="icon is-small"
+              title="Apply 'Include' filter"
+              @click="addFilter(attributeKey, params.data.ioc, 'must')"
+              ><i class="fas fa-search-plus"></i
+            ></span>
+            <pre>{{ params.data.ioc }}</pre>
+          </div>
+          <b-field v-if="!isInIntelligence(params.data)" grouped message="Add to local intelligence">
+            <b-select size="is-small" placeholder="IOC type" v-model="params.data.type">
+              <option v-for="option in IOCTypes" :value="option.type" :key="option.type">
+                {{ option.type }}
+              </option>
+            </b-select>
+            <b-button size="is-small" type="is-primary" @click="saveThreatIntel(params.data)">Add</b-button>
+          </b-field>
+          <div v-else>
+            <small>Already added to <router-link :to="{ name: 'Intelligence' }">Intelligence</router-link></small>
+          </div>
+        </section>
+      </template>
+    </TsContextMenu>
+  </div>
+</template>
+
+<script>
+import ApiClient from '../../utils/RestApiClient'
+import TsContextMenu from './TsContextMenu'
+import { SnackbarProgrammatic as Snackbar } from 'buefy'
+
+export default {
+  components: {
+    TsContextMenu,
+  },
+  props: ['attributeKey'],
+  name: 'TsIOCMenu',
+  data() {
+    return {
+      IOCTypes: [
+        { regex: /[0-9]{1,3}(\.[0-9]{1,3}\.)/g, type: 'ip' },
```

Comment:
No IPv6? :)

Reply:
Not yet :) I feel the reflex is a tad more complicated